### ISSUE TITLE
Ability to Load Extensions / Plugins from .eslintrc.yml?

### ISSUE THREAD

**Issue by joncursi** [codeclimate/codeclimate-eslint#87]
I have an `.eslintrc.yml` in the root of my project that pulls in a few rules from popular extensions and plugins:

```
---
plugins:
  - meteor
  - react
  - html
extends:
  - airbnb/base
  - plugin:react/recommended
```

Say I have this code:

```
App.propTypes = {
  blockchains: PropTypes.array.isRequired,
  bottomNavIcons: PropTypes.array.isRequired # should have a trailing comma based on airbnb/base plugin
};
```

Outside of CodeClimate CLI, ESLint will flag an error for not having the trailing comma on the third line (`comma-dangle`). This is expected behavior based on my setup. However, running `codeclimate analyze` does not catch the error. How do you load these extensions and plugins into codeclimate cli?

I have the following `.codeclimate.yml` config in the root of my project:

```
  eslint:
    enabled: true
    config:
      config: ./.eslintrc.yml
```

**Comment by wfleming** [comment]
Hi @joncursi,

Sorry for the delayed reply here.

Can you please provide the full content of your `.codeclimate.yml`? One thing to check would be that there aren't any `exclude_paths` in that yaml that might be excluding the file in question: any `exclude_paths` in the `.codecliamte.yml` are processed in addition to any engine-specific exclude paths in a file like `.eslintignore`.

As a more general explanation of how we load plugins in the CLI: we unfortunately do not currently support arbitrary ESLint plugins. Since plugins are effectively arbitrary remote code executing on our system, we need to be very careful about our architecture choices before we can support them. Right now we bundle & whitelist several plugins into the engine (`eslint-config-airbnb`, `eslint-plugin-react`, and `eslint-plugin-babel`), but other plugins won't work.

**Comment by joncursi** [comment]
Thanks for the info! Here's my `.codeclimate.yml`:

```
---
engines:
  duplication:
    enabled: true
    config:
      languages:
        javascript:
          mass_threshold: 200
    exclude_paths:
    - "**.test.js"
  eslint:
    enabled: true
    config:
      config: .eslintrc.yml
  fixme:
    enabled: true
    config:
      strings:
      - FIXME
      - TODO
      - BUG
  markdownlint:
    enabled: false
  nodesecurity:
    enabled: false
  requiresafe:
    enabled: false
  scss-lint:
    enabled: true
  shellcheck:
    enabled: true
ratings:
  paths:
  - "**.inc"
  - "**.es6"
  - "**.js"
  - "**.jsx"
  - "**.module"
exclude_paths:
- .meteor/
- node_modules/
- .codeclimate.yml
```

After some additional testing, I realized that the issue lies with CodeClimate not crawling `.jsx` files (where my comma dangle issues were located). Is there a way to enable that in the above `.codeclimate.yml`? Seems like it only picks up `.js`.

**Comment by wfleming** [comment]
> After some additional testing, I realized that the issue lies with CodeClimate not crawling .jsx files (where my comma dangle issues were located). Is there a way to enable that in the above .codeclimate.yml? Seems like it only picks up .js.

Ah, thanks, that's a good find. Yes, you can customize which extensions the engine will analyze: there are some docs at https://docs.codeclimate.com/docs/eslint#section-extensions.

**Comment by joncursi** [comment]
Thank you, that worked for me!
